kyaml/yaml/walk/associative_sequence.go

```diff
@@ -278,7 +223,7 @@ func (l *Walker) walkAssociativeSequence() (*yaml.RNode, error) {
 	}
 
 	// primitive associative list -- merge the values
```

Comment:
so `setPrimitiveSequenceElements` is dropped in favor of `setAssociativeSequenceElements`, and `elementValue` is dropped, leaving only `elementPrimitiveValue`.

Is there some point to the word _primitive_ any more?  I don't see it.

Reply:
There were three functions:
elementValue (used only by setPrimitiveSequenceElements)
elementValues - used to get values given a set of keys
elementPrimitiveValues - used to get values from a primitive list

setAssociativeSequenceElements and setPrimitiveSequenceElements were similar enough that a tweak to elementPrimitiveValues meant it could use setAssociativeSequenceElements instead

However, there is still a distinction between elementValues and elementPrimitiveValues that I think is important to keep